src/providers/NetworkProvider.tsx

```diff
@@ -109,24 +84,26 @@ export const NetworkProvider: React.FC = ({ children }) => {
     }
   }, [onboard])
 
-  // Refresh when network changes
-  useEffect(() => {
-    if (!previousNetwork || !network) return
-    if (previousNetwork === network) return
-    router.reload()
-  }, [network, previousNetwork, router])
+  const usingCorrectNetwork =
+    signingProvider?.network?.chainId === readNetwork.chainId
+
+  const walletIsReady = useCallback(async () => {
+    if (!userAddress || !onboard) {
+      await onSelectWallet()
+      return false
+    }
+    return await onboard.walletCheck()
+  }, [userAddress, onboard, onSelectWallet])
 
   return (
     <NetworkContext.Provider
       value={{
-        signerNetwork: network,
-        signingProvider:
-          signingProvider && network === readNetwork.name && account
-            ? signingProvider
-            : undefined,
-        userAddress: account,
-        onSelectWallet: selectWallet,
-        onLogOut: logOut,
+        shouldSwitchNetwork: signingProvider && !usingCorrectNetwork,
+        signingProvider: usingCorrectNetwork ? signingProvider : undefined,
```

Comment:
we ONLY define NetworkContext.signingProvider if using correct network.

this may not be completely intuitive, but it's a simple way to ensure ContractReaders don't try reading the wrong network, and that no transactions are created to the wrong network

Reply:
> This is one of those PRs where testing cross-browser def matters. I'm having issues with my non-brave wallets (on prod as well), if someone else could test maybe... 🥹

I can test tomorrow if y'all still need it.